README.md

```diff
@@ -1 +1,457 @@
+    },
+    {
+      "with": "as:did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp:*",
+      "can": "*"
+    }
+  ]
+}
+```
+
+## 3.3 Responder Establishes Point-to-Point Session
+
+**NOTE: The Responder is not yet trusted at this step, and MUST be treated as a possible impersonator or PITM**
+
+```
+Requestor                  Responder
+    ⋮                          ⋮
+    │       Authorization      │ (3a)
+    │       & Session Key      │ (3b)
+    │◄─────────────────────────┤
+    ⋮                          ⋮
+```
+
+In this step, the Responder MUST prove that they have access to the requested resources, and MUST set the input key material (IKM) that will be used to start a protected point-to-point connection. This is used to establish trust in the capabilities of the Responder, but MUST NOT actually delegating anything.
+
+The temporary RSA key from the previous step MUST be exclusively used for exchanging a 256-bit AES "session key". RSA is both slow and can only hold a limited number of bytes, so using it to encrypt the payloads of the rest of the session is infeasable.
+
+The payload contains two encryption layers, and signature: the RSA envelope, the AES envelope, and the AES key signed by the Responder.
+
+```
+          Payload
+
+        ┌───RSA───┐
+        │         │
+        │ AES-GCM │
+        │    │    │
+        └────┼────┘
+             │
+             │
+             ▼
+┌─────────AES-GCM────────┐
+│                        │
+│  ┌───────UCAN───────┐  │
+│  │                  │  │
+│  │  iss: Responder  │  │
+│  │  aud: TempDID    │  │
+│  │  fct: AES-GCM    │  │
+│  │  att: []         │  │
+│  │  prf: ...        │  │
+│  │                  │  │
+│  └──────────────────┘  │
+│                        │
+└────────────────────────┘
+```
+
+Upon receipt, the Requestor MUST validate that the UCAN capabilities fulfill their `caps` criterea. The UCAN itself MUST be valid, unrevoked, unexpired, and intended for the temporary DID (the `aud` field). If any of these checks fail, the session MUST be abandoned, the temporary DID regenerated, and the protocol restarted from [intention braodcast](#32-requestor-broadcasts-intent).
+
+### 3.3.1 Key Exchange
+
+The Responder MUST generate a fresh 256-bit AES key and 12-byte initialization vector (IV) per connection request. It is RECOMMENDED that the Responder track all DIDs requested with, and to only respond to each temporary DID exactly once.
+
+The key used in this step MUST be used as input key material (IKM) to derive keys in rest of the AWAKE bootstrap. The AES key MUST be encoded as padded base64 and included in the facts (`fct`) section of the payload UCAN. The rest of the [validation UCAN is constructed](#332-validation-ucan) and signed per normal, thus including the AES key in the signature payload. This signature serves as proof that the AES key was intended for this specific session.
+
+``` javascript
+{
+  fct: [
+    {
+      "awake/ikm": base64PaddedIkm,
```

Comment:
Why is this necessary if you already need the key to decrypt it?

Reply:
There's a reason, but this step is changing substantially due to the double ratchet so 🤷‍♀️